### PR TITLE
Implement column title alignment for Tree

### PR DESCRIPTION
**Pull request by Barugon**
godotengine/godot-proposals#982

Implement methods `Tree::set_column_title_alignment` and `Tree::get_column_title_alignment`.

### CODE REVIEW

**Comment by Barugon** [comment]
Results using the attached project:

Screenshot from 2023-02-25 13-21-05
[test_project.zip](https://github.com/godotengine/godot/files/10832405/test_project.zip)

File: scene/gui/tree.cpp [hunk 1]
```diff
@@ -4045,7 +4045,24 @@ void Tree::_notification(int p_what) {
 					int clip_w = tbrect.size.width - sb->get_minimum_size().width;
 					columns.write[i].text_buf->set_width(clip_w);
 
-					Vector2 text_pos = tbrect.position + Point2i(sb->get_offset().x + (tbrect.size.width - columns[i].text_buf->get_size().x) / 2, (tbrect.size.height - columns[i].text_buf->get_size().y) / 2);
+					Vector2 text_pos;
+					switch (columns[i].title_alignment) {
```
Comment (bruvzg) [inline scene/gui/tree.cpp:4049]: Left/Right should be swapped if `cache.rtl` is `true`.

**Comment by Barugon** [comment]
It seems to me that `HORIZONTAL_ALIGNMENT_LEFT` and `HORIZONTAL_ALIGNMENT_RIGHT` should be renamed to `HORIZONTAL_ALIGNMENT_BEGIN` and `HORIZONTAL_ALIGNMENT_END`, but that's a separate discussion.

**Comment by Barugon** [comment]
Any chance of getting this in v4.1?

**Comment by Barugon** [comment]
Windows failure doesn't look like it's due to my changes.

**Comment by KoBeWi** [comment]
I think you need to rebase the PR to fix it.

File: doc/classes/Tree.xml [hunk 2]
```diff
@@ -288,6 +295,14 @@
 				Sets the title of a column.
 			</description>
 		</method>
+		<method name="set_column_title_alignment">
+			<return type="void" />
+			<param index="0" name="column" type="int" />
+			<param index="1" name="title_alignment" type="int" enum="HorizontalAlignment" />
+			<description>
+				Sets column title alignment.
```
Comment (KoBeWi) [inline doc/classes/Tree.xml:303]: Seems like column titles don't support newlines, so `HORIZONTAL_ALIGNMENT_FILL` won't have effect here. Might be worth mentioning that maybe.

Comment (Barugon) [reply doc/classes/Tree.xml:303]: Should setting `HORIZONTAL_ALIGNMENT_FILL` produce an error message?

Comment (KoBeWi) [reply doc/classes/Tree.xml:303]: Maybe it could print a warning 🤔

Comment (Barugon) [reply doc/classes/Tree.xml:303]: Perhaps something like this?
```C++
	if (p_alignment == HORIZONTAL_ALIGNMENT_FILL) {
		WARN_PRINT("HORIZONTAL_ALIGNMENT_FILL is not supported for column titles");
		return;
	}
```

Comment (KoBeWi) [reply doc/classes/Tree.xml:303]: I think the warning should be in `set_text_alignment()` if anything.

**Review by KoBeWi — APPROVED** [review]
Well aside the note about fill alignment, seems fine. Needs squashing tho.

**Comment by Barugon** [comment]
> Needs squashing tho.

I don't know how to accomplish that.

**Comment by Barugon** [comment]
Well, that didn't work so good.

**Comment by Barugon** [comment]
I'm going to close this PR ad redo it.

**Comment by KoBeWi** [comment]
This is why you should never use the GitHub buttons to rebase xd

**Pull request closed without merging**